example/snippets.cpp

```diff
@@ -0,0 +1,805 @@
+    //]
+
+    //[snippet_modification_4
+    try
+    {
+        u.set_scheme( "100" ); // illegal, must start with a letter
+    }
+    catch( std::invalid_argument const& )
+    {
+        // handle error
+    }
+    //]
+
+    //[snippet_modification_5
+    u.set_host( parse_ipv4_address( "192.168.0.1" ).value() )
+        .set_port( 8080 )
+        .remove_userinfo();
+    //]
+
+    //[snippet_modification_6
+    params p = u.params();
+    p.emplace_at(p.find("name"), "name", "Vinnie Falco");
+    std::cout << u << "\n";
+    //]
+}
+
+void
+parsing_urls() {
+    //[snippet_parsing_url_1
+    result< url_view > r = parse_uri( "https://www.example.com/path/to/file.txt" );
+
+    if( r.has_value() )                         // parsing was successful
+    {
+        url_view u = r.value();                 // extract the url_view
+
+        std::cout << u;                         // format the URL to cout
+    }
+    else
+    {
+        std::cout << r.error().message();       // parsing failure; print error
+    }
+    //]
+
+    //[snippet_parsing_url_2
+    // This will hold our copy
+    std::shared_ptr<url_view const> sp;
+    {
+        std::string s = "/path/to/file.txt";
+
+        // result::value() will throw an exception if an error occurs
+        url_view u = parse_relative_ref( s ).value();
+
+        // create a copy with ownership and string lifetime extension
+        sp = u.collect();
+
+        // At this point the string goes out of scope
+    }
+
+    // but `*sp` remains valid since it has its own copy
+    std::cout << *sp << "\n";
+    //]
+
+    {
+        //[snippet_parsing_url_3
+        // This will hold our mutable copy
+        url v;
+        {
+            std::string s = "/path/to/file.txt";
+
+            // result::value() will throw an exception if an error occurs
+            v = parse_relative_ref(s).value();
+
+            // At this point the string goes out of scope
+        }
+
+        // but `v` remains valid since it has its own copy
+        std::cout << v << "\n";
+
+        // and it's mutable
+        v.set_encoded_fragment("anchor");
+
+        std::cout << v << "\n";
+        //]
+    }
+}
+
+void
+parsing_scheme() {
+    {
+        //[snippet_parsing_scheme_1
+        string_view s = "mailto:name@email.com";
+        url_view u = parse_uri( s ).value();
+        std::cout << u.scheme() << "\n";
+        //]
+    }
+    {
+        string_view s = "mailto:name@email.com";
+        //[snippet_parsing_scheme_2
+        url_view u = parse_uri( s ).value();
+        if (u.has_scheme())
+        {
+            std::cout << u.scheme() << "\n";
+        }
+        //]
+    }
+    {
+        //[snippet_parsing_scheme_3
+        string_view s = "file://host/path/to/file";
+        url_view u = parse_uri( s ).value();
+        if (u.scheme_id() == scheme::file)
+        {
+            // handle file
+        }
+        //]
+    }
+}
+
+void
+parsing_authority() {
```

Comment:
Why can't we get the braces right?

Reply:
I had got 89/95 braces right this time :D